store/dhash/dhash.go

```diff
@@ -62,9 +72,11 @@ func EncryptAES(payload, passphrase []byte) ([]byte, []byte, error) {
 	derivedKey := deriveKey([]byte(passphrase))
 
 	// Create initialization vector (nonse) to be used during encryption
-	// Nonce is derived from the mulithash (passpharase) so that encrypted payloads
-	// for the same multihash can be compared to each other without having to decrypt
-	nonce := SHA256(passphrase, nil)[:nonceLen]
+	// Nonce is derived from the passphrase concatenated with the payload so that the encrypted payloads
+	// for the same multihash can be compared to each other without having to decrypt them, as it's not possible.
+	payloadLen := make([]byte, 8)
+	binary.LittleEndian.PutUint64(payloadLen, uint64(len(payload)))
+	nonce := sha256Multiple(nil, noncePrefix, payloadLen, payload, passphrase)[:nonceLen]
 
 	// Create cypher and seal the data
 	block, err := aes.NewCipher(derivedKey)
```

Comment:
The format for Encrypted PeerID/Metadata was recently updated in the [DHT Spec](https://github.com/ipfs/specs/pull/373). The format should be [`encryption_varint`, `payload_len`, `nonce`, `encrypted_payload`]. See https://github.com/ipfs/specs/pull/373/#discussion_r1120001356

 The encryption varint for `aes-gcm-256` is `0x8040` and the multicodec is `0x2000` (See https://github.com/multiformats/multicodec/pull/314#issuecomment-1436537435)

Reply:
Great point! As IPNI doesn't have anything stored after `EncPeerID` the `encryption_varint` and `payload_len` can be appended on retrieval.

Reply:
Perfect, just wanted to make sure we are on the same page 😄